sys-utils/libcmt/src/io.h

```diff
@@ -0,0 +1,166 @@
+/* Copyright Cartesi and individual authors (see AUTHORS)
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file
+ * @defgroup libcmt_io_driver Cartesi Machine Input/Output Driver
+ * Low level abstraction of the Cartesi Machine kernel driver.
+ *
+ * Interaction is as follows:
+ * - Open the kernel device and mmap the shared memory ranges (tx and rx) with:
+ *   @ref cmt_io_init.
+ * - Retrieve the memory regions from the library with: @ref cmt_io_get_tx and
+ *   @ref cmt_io_get_rx.
+ * - Do requests (next input, emit output ...) by yielding the control back to
+ *   the emulator with: @ref cmt_io_yield.
+ *
+ * Lets look at some code:
+ *
+ * @include examples/io.c
+ *
+ * There are two implementations provided: `ioctl` that interacts with the
+ * cartesi-machine linux driver and `mock` that reads and writes the host
+ * filesystem, used for testing.
+ *
+ * @section Environment variables
+ *
+ * this module exposes two environment variables: @p CMT_DEBUG and @p CMT_INPUTS.
+ *
+ * @p CMT_DEBUG prints runtime information during application execution.
+ *
+ * ```
+ * CMT_DEBUG=yes ./application
+ * ```
+ *
+ * @p CMT_INPUTS feeds inputs to the mock (ignored for ioctl).
+ *
+ * ```
+ * CMT_INPUTS=0:advance.bin,1:inspect.bin ./application
+ * ```
+ *
+ * @ingroup libcmt
+ * @{ */
+#ifndef CMT_IO_H
+#define CMT_IO_H
+#include "buf.h"
+#include <stdint.h>
+
+/** Device */
+enum {
+    HTIF_DEVICE_YIELD = 2, /**< Yield device */
```

Comment:
I disagree with exposing a separate namespace for these symbols.
Users of the library shouldn't need to care about the fact that these are `HTIF` related.